lang/ghc/Portfile

```diff
@@ -118,26 +126,55 @@ if { [variant_isset "prebuilt"] } {
                     port:texlive-fonts-recommended \
                     port:texlive-latex-extra
 
-    pre-configure {
-        xinstall -m 0755 -d ${bootstrap_dir}
+    # Add these MacPorts dependencies after this issue is fixed:
+    # https://gitlab.haskell.org/ghc/ghc/-/issues/18752
+    # depends_lib-append \
+    #                 port:gmp \
+    #                 port:libffi \
+    #                 port:libiconv
+
+    post-extract {
+        # test: move extracted testsuite directory to src
+        xinstall -m 0755 -d ${srcpath}/${distname}
+        move        ${worksrcpath}/testsuite ${srcpath}/${distname}/testsuite
     }
 
+    # fix iconv linking issues if port libiconv is installed
+    # issue: file `[lindex [glob -directory ${worksrcpath}/libraries/base/dist-install/build -type f libHSbase-*-ghc${version}.dylib] 0]`
+    # links to /usr/lib/libiconv.dylib, but ${prefix}/lib/libiconv.dylib
+    # purposely mangles symbol names
+    set compiler_library_path_orig ${compiler.library_path}
+    compiler.library_path \
+                    ${compiler.library_path}:/usr/lib
```

Comment:
The compiler_library_path_orig variable doesn't seem to be used anywhere else. I would guess you don't need to add /usr/lib to compiler.library_path since I would expect the compiler to know to look there automatically.

Reply:
This line is absolutely necessary. I've checked, and `ghc` says that it uses `LIBRARY_PATH`.

Reply:
> I would expect the compiler to know to look there automatically.

It does.